rule.php

```diff
@@ -86,22 +86,19 @@ public static function add_settings_form_fields(
                     0 => get_string('notrequired', 'quizaccess_honestycheck'),
                     1 => get_string('honestycheckrequiredoption', 'quizaccess_honestycheck'),
                 ));
+        $default = get_config('quizaccess_honestycheck', 'honestycheckrequired');
+        $mform->setDefault('honestycheckrequired', $default);
         $mform->addHelpButton('honestycheckrequired',
                 'honestycheckrequired', 'quizaccess_honestycheck');
     }
 
     public static function save_settings($quiz) {
         global $DB;
-        if (empty($quiz->honestycheckrequired)) {
-            $DB->delete_records('quizaccess_honestycheck', array('quizid' => $quiz->id));
-        } else {
-            if (!$DB->record_exists('quizaccess_honestycheck', array('quizid' => $quiz->id))) {
-                $record = new stdClass();
-                $record->quizid = $quiz->id;
-                $record->honestycheckrequired = 1;
-                $DB->insert_record('quizaccess_honestycheck', $record);
-            }
-        }
+        $record = new stdClass();
+        $record->quizid = $quiz->id;
+        $record->honestycheckrequired = (empty($quiz->honestycheckrequired)) ? 0 : 1;
+        $DB->delete_records('quizaccess_honestycheck', ['quizid' => $record->quizid]);
+        $DB->insert_record('quizaccess_honestycheck', $record);
```

Comment:
Why did you change the existing code here? Your new code stores unnecsesary data in the DB. It was coded the way it was for a reason.

Reply:
Hi Tim,
This is what had triggered this work, this piece of code had resulted in silent failing of the setting to be saved when 'no required' was being selected on an activity. The empty check in here was taking the 0 from the 'no required' option as empty and deleting the record. 

I have updated the code to not store unnecessary data but update the record if it already exists.